src/App.tsx

```diff
@@ -4,32 +4,71 @@ import logo from "../public/FischLogo.svg";
 
 import styles from "./App.module.css";
 import { Card } from "./components/Card";
-import { useRef, useState } from "react";
+import { JSX, useEffect, useState } from "react";
+import { NewCard } from "./components/NewCard";
 
-function App() {
-  const [cardsAmount, setCardsAmount] = useState(1)
-  const [cards, setCards] = useState([{}]);
+interface AppProps {
+  basicCards: {
+    face: string;
+    back: string;
+  }[];
+}
+
+interface CardObject {
+  face: string;
+  back: string;
+}
+
+function App(props: AppProps) {
+  const importedCards = props.basicCards.map((card, index) => {
+    return { face: card.face, back: card.back, key: index };
+  });
+
+  const [cardsArray, setCardsArray] = useState(importedCards);
+
+  const [incrementalKey, setIncrementalKey] = useState(cardsArray.length);
+
+  useEffect(
+    () => setIncrementalKey((prevKey) => prevKey + 1),
+    [cardsArray, setCardsArray]
+  );
+
+  const [isNewCardDisplayed, setIsNewCardDisplayed] = useState(false);
 
   const handleAddCardButtonClick = () => {
-    setCardsAmount(cardsAmount+1)
-    setCards(prevCards => [...prevCards, {}]);
+    setIsNewCardDisplayed(true);
+  };
+
+  const handleSaveButtonClick = (props: CardObject) => {
+    setCardsArray([
+      { face: props.face, back: props.back, key: incrementalKey },
+      ...cardsArray,
+    ]);
+    setIsNewCardDisplayed(false);
   };
-  
 
-  const renderCards = () => {
-    return cards.map(() => (
-      <Card cardsAmount={cardsAmount} setCardsAmount={setCardsAmount} />
-    ));
+  const handleCancelButtonClick = () => {
+    setIsNewCardDisplayed(false);
   };
 
   return (
     <AppLayout>
       <AppHeader
-        cardsAmount={cardsAmount}
+        cardsAmount={cardsArray.length + (isNewCardDisplayed ? 1 : 0)}
```

Comment:
**nitpick:** Why so complicated? It should be just:
```jsx
 cardsAmount={cardsArray.length}
```

Reply:
This one I'm gonna stand by, because it's specially indicated in project requirements that this is how it should work: 

![Screenshot 2023-07-18 at 20 57 12](https://github.com/filwas/fischkapp/assets/69075808/00e2e054-4821-4f02-8ba7-2339d23780b0)

NewCard is not inside cardsArray, therefore clicking "add new card" wouldn't actually increase the counter until we saved it.

Reply:
Ok, you are right - I had to ask Karolina - it's a new requirement in this updated design :D